cli/server_test.go

```diff
@@ -0,0 +1,195 @@
+	require.NoError(t, err, "could not load config")
+	ptr := &goodCfg
+	saveCfg := func(t *testing.T, f func(cfg *config.Config)) string {
+		cfg := *ptr
+		chainPath := filepath.Join(t.TempDir(), "neogotestchain")
+		cfg.ApplicationConfiguration.DBConfiguration.Type = "leveldb"
+		cfg.ApplicationConfiguration.DBConfiguration.LevelDBOptions.DataDirectoryPath = chainPath
+		f(&cfg)
+		out, err := yaml.Marshal(cfg)
+		require.NoError(t, err)
+
+		cfgPath := filepath.Join(tmpDir, "protocol.unit_testnet.yml")
+		require.NoError(t, ioutil.WriteFile(cfgPath, out, os.ModePerm))
+		t.Cleanup(func() {
+			require.NoError(t, os.Remove(cfgPath))
+		})
+		return cfgPath
+	}
+
+	baseCmd := []string{"neo-go", "node", "--unittest", "--config-path", tmpDir}
+	e := newExecutor(t, false)
+
+	t.Run("invalid config path", func(t *testing.T) {
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("bad logger config", func(t *testing.T) {
+		badConfigDir := t.TempDir()
+		logfile := filepath.Join(badConfigDir, "logdir")
+		require.NoError(t, os.WriteFile(logfile, []byte{1, 2, 3}, os.ModePerm))
+		saveCfg(t, func(cfg *config.Config) {
+			cfg.ApplicationConfiguration.LogPath = filepath.Join(logfile, "file.log")
+		})
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("invalid storage", func(t *testing.T) {
+		saveCfg(t, func(cfg *config.Config) {
+			cfg.ApplicationConfiguration.DBConfiguration.Type = ""
+		})
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("stateroot service is on && StateRootInHeader=true", func(t *testing.T) {
+		saveCfg(t, func(cfg *config.Config) {
+			cfg.ApplicationConfiguration.StateRoot.Enabled = true
+			cfg.ProtocolConfiguration.StateRootInHeader = true
+		})
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("invalid Oracle config", func(t *testing.T) {
+		saveCfg(t, func(cfg *config.Config) {
+			cfg.ApplicationConfiguration.Oracle.Enabled = true
+			cfg.ApplicationConfiguration.Oracle.UnlockWallet.Path = "bad_orc_wallet.json"
+		})
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("invalid consensus config", func(t *testing.T) {
+		saveCfg(t, func(cfg *config.Config) {
+			cfg.ApplicationConfiguration.UnlockWallet.Path = "bad_consensus_wallet.json"
+		})
+		e.RunWithError(t, baseCmd...)
+	})
+	t.Run("invalid Notary config", func(t *testing.T) {
+		t.Run("malformed config", func(t *testing.T) {
+			saveCfg(t, func(cfg *config.Config) {
+				cfg.ProtocolConfiguration.P2PSigExtensions = false
+				cfg.ApplicationConfiguration.P2PNotary.Enabled = true
+			})
+			e.RunWithError(t, baseCmd...)
+		})
+		t.Run("invalid wallet", func(t *testing.T) {
+			saveCfg(t, func(cfg *config.Config) {
+				cfg.ProtocolConfiguration.P2PSigExtensions = true
+				cfg.ApplicationConfiguration.P2PNotary.Enabled = true
+				cfg.ApplicationConfiguration.P2PNotary.UnlockWallet.Path = "bad_notary_wallet.json"
+			})
+			e.RunWithError(t, baseCmd...)
+		})
+	})
+	t.Run("good", func(t *testing.T) {
+		/*
+			// TODO: how to check it?
```

Comment:
Theoretically, by checking it via RPC, but it won't be convenient for sure and there is a question of how to stop it. Maybe it's not the most critical thing for us now (although checking shutdown procedure would be nice too).

Reply:
I think, sending `os.Interrupt` to our process might work (this requires executing `Run` in a separate goroutine).

Reply:
A made the test with RPC, it works fine. But currently, the node is not shutting down properly, only when the whole test ends. IIUC, `os.Interrupt` can't be handled on windows.